Make some miscellaneous fixes

jsuhaas22 opened the pull request:
This PR makes a bunch of little changes:

- It fixes the `shutdown now` button, which at present (before this PR) doesn't shutdown the system
- It uses a questionable hack to get around a couple of issues in ARM Analytics
- It does some cleanups

cshilwant commented:
@glneo I see other methods like Component.Ready which probably can be looked into? - https://doc.qt.io/qt-6/qml-qtqml-component-members.html

glneo commented:
Hmm, seems there is a signal from `GstGLQt6VideoItem` called `onItemInitializedChanged` when the context is ready. This works on my desktop, could you try the following in board:

```
diff --git a/apps/arm_analytics.qml b/apps/arm_analytics.qml
index 1648b1f..85868d1 100644
--- a/apps/arm_analytics.qml
+++ b/apps/arm_analytics.qml
@@ -43,13 +43,12 @@ Rectangle {
                 }
             }
         }
-    }
 
-    Component.onCompleted: {
-//        arm_analytics.startVideo(analytics_feed, "Object Detection");
-    }
-    Component.onDestruction: {
-        arm_analytics.stopVideo();
+        onItemInitializedChanged: {
+            arm_analytics.startVideo(analytics_feed, "Object Detection");
+        }
+        Component.onDestruction: {
+            arm_analytics.stopVideo();
+        }
     }
 }
-

jsuhaas22 commented:
@glneo Your change worked! Have updated the PR.

cshilwant commented:
> @glneo Your change worked! Have updated the PR.

@jsuhaas22 you may also need to update commit message for 86f56a0 ?

As we aren't doing `Run a pipeline in Component.onDestruction` anymore

jsuhaas22 commented:
> @jsuhaas22 you may also need to update commit message for [86f56a0](https://github.com/TexasInstruments/ti-apps-launcher/commit/86f56a05f7cf91ae586b374e617fff7f209a235b) ?
> 
> As we aren't doing `Run a pipeline in Component.onDestruction` anymore

Thanks @cshilwant , had missed that. Updated the commit message.

Further, @glneo and @cshilwant : I have added a new commit to fix a new problem I discovered now. Please re-review.

cshilwant reviewed (approved):
LGTM!
